src/main/common/crc.h

```diff
@@ -27,8 +27,14 @@ uint16_t crc16_ccitt_update(uint16_t crc, const void *data, uint32_t length);
 struct sbuf_s;
 void crc16_ccitt_sbuf_append(struct sbuf_s *dst, uint8_t *start);
 
-uint8_t crc8_dvb_s2(uint8_t crc, unsigned char a);
-uint8_t crc8_dvb_s2_update(uint8_t crc, const void *data, uint32_t length);
-void crc8_dvb_s2_sbuf_append(struct sbuf_s *dst, uint8_t *start);
+uint8_t crc8_calc(uint8_t crc, unsigned char a, uint8_t poly);
+uint8_t crc8_update(uint8_t crc, const void *data, uint32_t length, uint8_t poly);
+void crc8_sbuf_append(struct sbuf_s *dst, uint8_t *start, uint8_t poly);
+#define crc8_dvb_s2(crc, a)                         crc8_calc(crc, a, 0xD5)
+#define crc8_dvb_s2_update(crc, data, length)       crc8_update(crc, data, length, 0xD5)
+#define crc8_dvb_s2_sbuf_append(dst, start)         crc8_sbuf_append(dst, start, 0xD5)
+#define crc8_poly_0xba(crc, a)                      crc8_calc(crc, a, 0xBA)
+#define crc8_poly_0xba_sbuf_append(dst, start)      crc8_sbuf_append(dst, start, 0xBA)
```

Comment:
I wonder if there is a better crc choice here? 0xBA in Koopman notation is 0xDD, and it does not show up on the summary tables.

https://users.ece.cmu.edu/~koopman/crc/

(not a crc expert and I could be completely wrong)

Reply:
maybe 0xBA is not a good choice. but we use it in all CRSF command frames so we cannot change it.